vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/StaffSearchManager.java

```diff
@@ -0,0 +1,221 @@
+        int slidePageNo;
+        try {
+            slidePageNo = validatePageNumber(page, totalSlidePage);
+        } catch (NumberFormatException numberFormatException) {
+            slidePageNo = 1;
+        }
+        // if it is not the current tab then reset the page number to 1.
+        if (!tab.trim().equals("slide")) {
+            slidePageNo = 1;
+        }
+        strSlidePageNo.setLength(0);
+        strSlidePageNo.append(String.valueOf(slidePageNo));
+        Pageable requestedPageForSlide = PageRequest.of(slidePageNo - 1, pageSize);
+        HashSet<IVSpaceElement> resultElements = new LinkedHashSet<>();
+        resultElements.addAll(slideManager.findInNameOrDescription(requestedPageForSlide, search).getContent());
+        return resultElements;
+    }
+    
+    /**
+     * Method to return the requested slides whose text blocks contains the search string
+     * 
+     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
+     *                page is returned
+     * @return list of slides whose text blocks contains the search string in the requested pageNo and requested order.
+     */
+    @Override
+    public HashSet<IVSpaceElement> searchSlideTexts(String search, int page, long totalSlideTextPage, String tab,
+            StringBuilder strSlideTextPageNo) {
+        int slideTextPageNo;
+        try {
+            slideTextPageNo = validatePageNumber(page, totalSlideTextPage);
+        } catch (NumberFormatException numberFormatException) {
+            slideTextPageNo = 1;
+        }
+        // if it is not the current tab then reset the page number to 1.
+        if (!tab.trim().equals("slideText")) {
+            slideTextPageNo = 1;
+        }
+        strSlideTextPageNo.setLength(0);
+        strSlideTextPageNo.append(String.valueOf(slideTextPageNo));
+        Pageable requestedPageForSlideText = PageRequest.of(slideTextPageNo - 1, pageSize);
+        HashSet<IVSpaceElement> resultElements = new LinkedHashSet<>();
+        resultElements.addAll(textContentBlockRepo.findInNameOrDescription(requestedPageForSlideText, search).getContent());
+        return resultElements;
+    }
+
+    /**
+     * Method to return page number after validation
+     * 
+     * @param pageNo page provided by calling method
+     * @return 1 if pageNo less than 1 and lastPage if pageNo greater than
+     *         totalPages.
+     */
+    @Override
+    public int validatePageNumber(int pageNo, long totalPages) {
+        if (pageNo < 1) {
+            return 1;
+        } else if (pageNo > totalPages) {
+            return (totalPages == 0) ? 1 : (int) totalPages;
+        }
+        return pageNo;
+    }
+
+    /**
+     * Method to return the total pages sufficient to display all spaces/modules/slides/Slides text block
+     * 
+     * @return totalPages required to display all spaces/modules/slides/Slides text block in DB
+     */
+
+    @Override
+    public long getTotalPages(long count) {
```

Comment:
this method should not get any parameter; also since we have four different searches, there need to be four different gettotal methods.